students/EricAdams/session03/list_lab/list_lab.py

```diff
@@ -0,0 +1,149 @@
+# Python uses zero-based indexing, so you will need to correct.
+number = int(input(" Enter a number from 1 to " + str(len(fruits)) + "> "))
+print_what_was_entered(number, fruits)
+# Add another fruit to the beginning of the list using “+” and display the
+# list.
+answer = input("enter a fruit to add to the beginning of the list> ")
+fruits = [answer.strip().title()] + fruits
+display_list(fruits)
+# Add another fruit to the beginning of the list using insert() and display
+# the list.
+answer = input("> enter a fruit to add to the beginning of the list> ")
+fruits.insert(0, answer.strip().title())
+display_list(fruits)
+# Display all the fruits that begin with “P”, using a for loop.
+print("The following fruits begin with P:")
+for x in fruits:
+    if x[0] == 'P':
+        print(x)
+
+# Series 2
+print("\nSeries 2\n")
+fruits = fruits_orig[:]
+# Using the list created in series 1 above:
+# Display the list.
+display_list(fruits)
+# Remove the last fruit from the list.
+print("Removing the last fruit from the list")
+fruits.remove(fruits[-1])
+# Display the list.
+display_list(fruits)
+# Ask the user for a fruit to delete and find it and delete it.
+answer = input("Enter the fruit you want to be deleted.> ")
+answer = answer.strip().title()
+print("You chose {}".format(answer))
+i = 0
+for x in fruits:
+    if x == answer:
+        fruits.remove(fruits[i])
+    i += 1
+display_list(fruits)
+# (Bonus: Multiply the list times two. Keep asking until a match is found.
+# Once found, delete all occurrences.)
+print("Multipying the list by two")
+fruits = fruits * 2
+display_list(fruits)
+statement = "Enter the fruit you want to be deleted."
+while True:
+    answer = input(statement)
+    answer = answer.strip().title()
+    print("You chose {}".format(answer))
+    i = 0
+    match = False
+    for x in fruits:
+        if x == answer:
+            fruits.remove(fruits[i])
+            match = True
+        i += 1
+    if match is True:
+        break
+    else:
+        statement = answer + ' not found.  Try again'
+display_list(fruits)
+
+
+# Series 3
+print("\n Series 3\n")
+# Again, using the list from series 1:
+fruits = fruits_orig[:]
+display_list(fruits)
+# Ask the user for input displaying a line like “Do you like apples?” for each
+# fruit in the list (making the fruit all lowercase).
+i = 0
+for x in fruits:
+    answer = input('Do you like {}> '.format(x.lower()))
+    # For each “no”, delete that fruit from the list.
+    if answer.strip().lower() == 'no':
+        print('Deleting {} from the list'.format(x.lower()))
+        counter = 0
+        for y in fruits:
+            if fruits[counter] == x:
+                fruits.remove(x)
+    elif answer.strip().lower() == 'yes':
+        continue
+    # For any answer that is not “yes” or “no”, prompt the user to answer
+    # with one of those two values (a while loop is good here)
+    else:
+        while True:
```

Comment:
this wan't supposed to be in its own loop, but rather, part of the loop above :-)